session.go

```diff
@@ -89,41 +175,24 @@ func (s Session) SubredditSubmissions(subreddit string) ([]*Submission, error) {
 	return submissions, nil
 }
 
-// SortedSubmissions will return submissions from a subreddit (or homepage if "") by popularity and age
-// TODO Review this
-func (s Session) SortedSubmissions(subreddit string, popularity popularitySort, age ageSort) ([]*Submission, error) {
-	if age != DefaultAge {
-		switch popularity {
-		case NewSubmissions, RisingSubmissions, HotSubmissions:
-			return nil, fmt.Errorf("cannot sort %s by %s", popularity, age)
+// Returns top submmissions from a subreddit
+// Second argument must have value of one of "hour", "day", "week", "month", "year", "all"
+func (s Session) TopSubmissions(subreddit string, t PostAge) ([]*Submission, error) {
+	times := []string{"hour", "day", "week", "month", "year", "all"}
+	url := ""
+	timeValid := ""
+
+	//check if time is valid value
+	for _, k := range times {
```

Comment:
You don't need this loop. If you do not export the `PostAge` type (i.e. `type postAge string`), but export the possible values, (i.e. `const Hour postAge = "hour"`), there are no possible ways of creating more of the type `reddit.postAge` besides the ones exported by this library, thus it becomes impossible for a user of this method to pass an invalid `postAge`.